osquery/utils/azure/azure_util.cpp

```diff
@@ -0,0 +1,105 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <boost/property_tree/json_parser.hpp>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/remote/http_client.h>
+#include <osquery/utils/azure/azure_util.h>
+
+namespace pt = boost::property_tree;
+namespace http = osquery::http;
+namespace fs = boost::filesystem;
+
+namespace osquery {
+
+static bool isAzureInstance() {
+  static std::atomic<bool> checked(false);
+  static std::atomic<bool> is_azure_instance(false);
+
+  if (checked) {
+    return is_azure_instance;
+  }
+
+  static std::once_flag once_flag;
+  std::call_once(once_flag, []() {
+    if (checked) {
+      return;
+    }
+
+    checked = true;
+
+#ifdef WINDOWS
```

Comment:
Do you mind using `isPlatform(PlatformType::TYPE_WINDOWS)` / etc instead of preprocessor macros?